examples/summarizePatches.py

```diff
@@ -0,0 +1,179 @@
+                if patchBBox.overlaps(skymapBBoxes[skymapId]):
+                    visitId, ccdId = skymapId
+                    if visitId not in visitsToCcds:
+                        visitsToCcds[visitId] = [ccdId]
+                        visitFilter = butler.queryMetadata('calexp_md', 'filter', visit=visitId)[0]
+                        if visitFilter not in filtersToVisits:
+                            filtersToVisits[visitFilter] = []
+                        filtersToVisits[visitFilter] += [visitId]
+                        numOverlaps += 1
+                    else:
+                        visitsToCcds[visitId] += [ccdId]
+                        numOverlaps += 1
+
+            # If there was an overlap, print and save information
+            if numOverlaps != 0:
+                print("patch=%r:    Found %r overlapping ccds." % (patchId, numOverlaps))
+
+                if saveOverview:
+                    overview.write("patch=" + str(patchId) + "\n")
+                    overview.write("    " + str(filtersToVisits) + "\n")
+
+                if saveSummaries is not None:
+                    for i_f, patchFilter in enumerate(filtersToVisits):
+
+                        # Write expanded file - one line for each filter in each patch
+                        expandedSummary.write("filter=" + patchFilter + " ")
+                        expandedSummary.write("tract=" + str(tract.getId()) + " ")
+                        expandedSummary.write("patch=")
+                        expandedSummary.write(str(patchId[0]) + "," + str(patchId[1]) + " ")
+                        for i_v, visit in enumerate(filtersToVisits[patchFilter]):
+                            if i_v != 0:
+                                expandedSummary.write("^" + str(visit))
+                            else:
+                                expandedSummary.write("--selectId visit=" + str(visit))
+                        expandedSummary.write("\n")
+
+                        # Write condensed file - one line for each patch
+                        if i_f != 0:
+                            condensedSummary.write("^" + patchFilter)
+                        else:
+                            condensedSummary.write("filter=" + patchFilter)
+                    condensedSummary.write(" tract=" + str(tract.getId()) + " ")
+                    condensedSummary.write("patch=")
+                    condensedSummary.write(str(patchId[0]) + "," + str(patchId[1]) + " ")
+                    condensedSummary.write("\n")
+
+                # Save an overview of ccd/patch overlaps
+                if saveOverview:
+                    for visit in visitsToCcds:
+                        overview.write("        visit=" + str(visit) + "\n")
+                        for ccd in visitsToCcds[visit]:
+                            overview.write("            ccd=" + str(ccd) + "\n")
+                    overview.write("\n")
+
+def splitId(argName):
+    class SplitIdValueAction(IdValueAction):
```

Comment:
This is both clever and a terrible hack that I wish we didn't need, but I know from experience how hard it is to extract the Id parsing from pipe_base. It might be worth filing a ticket on pipe_base to make an easy to use version of the id parsing code, even though I'm not sure anyone is going to pick that up soon. There are definitely many other scripts like this that aren't suited to being full command line tasks but which need bits of the task infrastructure.

To be clear, I think this code is acceptable as-is.